mesos/cli/cmds/frameworks.py

```diff
@@ -0,0 +1,144 @@
+
+    def _lookup(self, state_count, state):
+        if state in state_count.keys():
+            return state_count[state]
+        else:
+            return 0
+
+
+def resource_stats(resource, allocated, used):
+    r_used = used[resource]
+    r_allocated = allocated[resource]
+    if r_allocated > 0:
+        percent = int(r_used / r_allocated * 100)
+        return str(r_used) + "/" + str(r_allocated) + "(" + str(percent) + "%)"
+    else:
+        # no resources are being used/allocated
+        return "-"
+
+@cli.init(parser)
+def main(args):
+    term = blessings.Terminal()
+    active_task_stats = TaskStats()
+    terminal_task_stats = TaskStats()
+
+    table_generator = OrderedDict([
+        ("ID", lambda x: x["id"]),
+        ("name", lambda x: x["name"]),
+        ("host", lambda x: x["hostname"]),
+        ("active", lambda x: x["active"]),
+        ("cpu", lambda x: resource_stats("cpus", x["resources"], x["used_resources"])),
+        ("mem", lambda x: resource_stats("mem", x["resources"], x["used_resources"])),
+        ("disk", lambda x: resource_stats("disk", x["resources"], x["used_resources"])),
+    ])
+
+    # TODO: the task breakdown is not done in an efficient way (since we traverse the
+    # task list on every framework). This can be optimized to do the traversal only
+    # once per framework.
+    if args.task_breakdown:
+        # active tasks
+        table_generator['staging'] = lambda x: active_task_stats.get(x["id"], x["tasks"], "TASK_STAGING")
+        table_generator['starting'] = lambda x: active_task_stats.get(x["id"], x["tasks"], "TASK_STARTING")
+        table_generator['running'] = lambda x: active_task_stats.get(x["id"], x["tasks"], "TASK_RUNNING")
+        # terminal state tasks. The count here is approximate since the API caps the result
+        # at a max of 1000 tasks
+        table_generator['failed'] = lambda x: terminal_task_stats.get(x["id"], x["completed_tasks"], "TASK_FAILED")
+        table_generator['killed'] = lambda x: terminal_task_stats.get(x["id"], x["completed_tasks"], "TASK_FAILED")
+        table_generator['error'] = lambda x: terminal_task_stats.get(x["id"], x["completed_tasks"], "TASK_ERROR")
+        table_generator['lost'] = lambda x: terminal_task_stats.get(x["id"], x["completed_tasks"], "TASK_ERROR")
+        table_generator['finished'] = lambda x: terminal_task_stats.get(x["id"], x["completed_tasks"], "TASK_FINISHED")
+    else:
+        table_generator['active tasks'] = lambda x: len(x["tasks"])
+
+    tb = prettytable.PrettyTable(
+        [x.upper() for x in table_generator.keys()],
+        border=False,
+        max_table_width=term.width,
+        hrules=prettytable.NONE,
+        vrules=prettytable.NONE,
+        left_padding_width=0,
+        right_padding_width=1
+    )
+
+    def format_row(framework):
+        return [fn(framework) for fn in table_generator.values()]
+
+    for framework in MASTER.state['frameworks']:
```

Comment:
This seems like an abuse of the API. You should be using `MASTER.frameworks(active_only=args.inactive)`.